pydantic/fields.py

```diff
@@ -30,6 +30,14 @@
 
 Required: Any = Ellipsis
 
+
+class UndefinedClass:
```

Comment:
```suggestion
class UndefinedType:
```
perhaps more canonical?

Reply:
Agreed, I wasn't happy with the name either, I like `UndefinedType` better.